### PR TITLE
add CI workflow

### PR DESCRIPTION
**Pull request by cmatheson**
Makes sure everything is formatted properly with prettier.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    name: Pre-merge Checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Setup
+        run: npm ci
```
Comment (jasonroelofs) [inline .github/workflows/ci.yml:20]: You do need to explicitly `npm install`. Doesn't look like `npm ci` actually does anything.

Comment (cmatheson) [reply .github/workflows/ci.yml:20]: `npm ci` [should do the install](https://docs.npmjs.com/cli/v10/commands/npm-ci) (i think i hadn't properly formatted all these files yet though).

Comment (jasonroelofs) [reply .github/workflows/ci.yml:20]: Ah, interesting, as the previous CI failures were saying that `prettier` wasn't installed. But it's passing now!

**Review by jasonroelofs — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cmatheson) [reply .github/workflows/ci.yml:20]: oooh, yeah that was a legit failure, i hadn't added prettier to the package.json yet 🤦

**Pull request merged**